### ISSUE TITLE
Can't decrypt secret key

### ISSUE THREAD

**Issue by IvanPizhenko** [pgpainless/pgpainless#154]
There is failed attempt to decrypt certain secret keys which are known to be correct.

Sample key and passphrase can be found here https://github.com/FlowCrypt/flowcrypt-android/issues/1296#issuecomment-877848671

For more information, please read through comments in the https://github.com/FlowCrypt/flowcrypt-android/issues/1296

Sample code:
```kotlin
  fun decryptKey(key: PGPSecretKeyRing, passphrase: Passphrase): PGPSecretKeyRing {
    return PGPainless.modifyKeyRing(key)
      .changePassphraseFromOldPassphrase(passphrase)
      .withSecureDefaultSettings()
      .toNoPassphrase()
      .done()
  }
```

My test which invokes this method with that key, produces an exception:
```
java.lang.NullPointerException
	at org.bouncycastle.openpgp.PGPSecretKey.buildSecretKeyPacket(Unknown Source)
	at org.bouncycastle.openpgp.PGPSecretKey.<init>(Unknown Source)
	at org.pgpainless.key.modification.secretkeyring.SecretKeyRingEditor.lockPrivateKey(SecretKeyRingEditor.java:625)
	at org.pgpainless.key.modification.secretkeyring.SecretKeyRingEditor.changePassphrase(SecretKeyRingEditor.java:597)
	at org.pgpainless.key.modification.secretkeyring.SecretKeyRingEditor.access$300(SecretKeyRingEditor.java:69)
	at org.pgpainless.key.modification.secretkeyring.SecretKeyRingEditor$WithPassphraseImpl.toNoPassphrase(SecretKeyRingEditor.java:579)
	at com.flowcrypt.email.security.pgp.PgpKey.decryptKey(PgpKey.kt:96)
	at com.flowcrypt.email.security.pgp.PgpKeyTest.testParseAndDecryptKey_Issue1296(PgpKeyTest.kt:138)
        ..... more stack frames cut off
```

**Comment by vanitasvitae** [comment]
I think I have an idea what might cause this.
I'll investigate.

**Comment by vanitasvitae** [comment]
Should be fixed with #155
